File: 238. Product of Array Except Self/solution.go

func productExceptSelf(nums []int) []int {
	ans := make([]int, len(nums))
	totalProduct := 1
	ans[0] = 1
	for i := 0; i < len(nums)-1; i++ {
		totalProduct = totalProduct * nums[i]
		ans[i+1] = totalProduct
	}
	totalProduct = 1
	for i := len(nums) - 1; i > 0; i-- {
		totalProduct = totalProduct * nums[i]
		ans[i-1] = ans[i-1] * totalProduct
	}
	return ans
}

File: 238. Product of Array Except Self/solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = [1] * len(nums)
        totalProduct = 1
        for i in range(len(nums)-1):
            totalProduct = totalProduct*nums[i]
            ans[i+1] = totalProduct
        totalProduct = 1
        for i in range(len(nums)-1, 0, -1):
            totalProduct = totalProduct*nums[i]
            ans[i-1] = ans[i-1]*totalProduct
        return ans
